SpatialGDK/Source/SpatialGDKServices/Private/LocalReceptionistProxyServerManager.cpp

```diff
@@ -0,0 +1,265 @@
+	}
+	else
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Loading proxyInfo.json failed. Can't get proxy's Pid."));
+	}
+
+	return nullptr;
+}
+
+void FLocalReceptionistProxyServerManager::SetPidInJson(const FString& Pid)
+{
+	FString ProxyInfoFileResult;
+
+	// If file does not exist create an empty json file 
+	if (!FPaths::FileExists(ProxyInfoFilePath))
+	{
+		FFileHelper::SaveStringToFile(TEXT("{ }"), *ProxyInfoFilePath);
+	}
+
+	TSharedPtr<FJsonObject> JsonParsedProxyInfoFile = ParsePidFile();
+	if (!JsonParsedProxyInfoFile.IsValid())
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Failed to update Pid(%s). Please ensure that the following file exists: %s"), *Pid, *SpatialGDKServicesConstants::ProxyInfoFilename);
+		return;
+	}
+
+	JsonParsedProxyInfoFile->SetStringField("pid", Pid);
+
+	TSharedRef<TJsonWriter<>> JsonWriter = TJsonWriterFactory<>::Create(&ProxyInfoFileResult);
+	if (!FJsonSerializer::Serialize(JsonParsedProxyInfoFile.ToSharedRef(), JsonWriter))
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Failed to write pid to parsed proxy info file. Unable to serialize content to json file."));
+		return;
+	}
+	if (!FFileHelper::SaveStringToFile(ProxyInfoFileResult, *FPaths::Combine(SpatialGDKServicesConstants::ProxyFileDirectory, SpatialGDKServicesConstants::ProxyInfoFilename)))
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("Failed to write file content to %s"), *SpatialGDKServicesConstants::ProxyInfoFilename);
+	}
+}
+
+FString FLocalReceptionistProxyServerManager::ParsePid()
+{
+	FString Pid;
+
+	if (TSharedPtr<FJsonObject> JsonParsedProxyInfoFile = ParsePidFile())
+	{
+		if (JsonParsedProxyInfoFile->TryGetStringField(TEXT("pid"), Pid))
+		{
+			return Pid;
+		}
+		else
+		{
+			UE_LOG(LogLocalReceptionistProxyServerManager, Error, TEXT("'pid' does not exist in proxyInfo.json. Can't read proxy's Pid."));
+		}
+	}
+
+	Pid.Empty();
+	return Pid;
+}
+
+bool FLocalReceptionistProxyServerManager::TryStartReceptionistProxyServer(bool bIsRunningInChina, const FString& CloudDeploymentName, const FString& ListeningAddress, const int32 ReceptionistPort)
+{
+	FString StartResult;
+	int32 ExitCode;
+
+	// Do not restart the same proxy if you have already a proxy running for the same cloud deployment
+	if (bProxyIsRunning && ProxyServerProcHandle.IsValid() && RunningCloudDeploymentName == CloudDeploymentName && RunningProxyListeningAddress == ListeningAddress && RunningProxyReceptionistPort == ReceptionistPort)
+	{
+		UE_LOG(LogLocalReceptionistProxyServerManager, Log, TEXT("The local receptionist proxy server is already running!"));
+
+		return true;
+	}
+
+	// Stop receptionist proxy server if it is for a different cloud deployment, port or listening address
+	if (bProxyIsRunning && ProxyServerProcHandle.IsValid())
+	{
+		if (!TryStopReceptionistProxyServer())
+		{
+			UE_LOG(LogLocalReceptionistProxyServerManager, Log, TEXT("Failed to stop previous proxy server!"));
```

Comment:
might want to make this a warning or an Error. Can we give the user any more information as to why it failed?